### PR TITLE
added structured datatypes to M3, serialization

### PR DESCRIPTION
**Pull request by enikao**
Value types: #265 

also updates specs to version 2024.1, removes JSON primitive type (#277)

### CODE REVIEW

**Comment by enikao** [comment]
Draft until we decided on structured datatype name.

File: bulk/repo-access-api.adoc [hunk 1]
```diff
@@ -5,11 +5,12 @@ include::../shared/issue-footnotes.adoc[]
 :arch: ../reference-architecture/reference-architecture
 :chunk: <<{serialization}.adoc#SerializationChunk, SerializationChunk>>
 
+[[bulk]]
 = Bulk Repository Access API
 :toc: preamble
 :toclevels: 2
 
-The bulk API is used to store and retrieve nodes in batches at the moment of invocation.{fn-org25}
+The bulk API *version 2024.1* is used to store and retrieve nodes in batches at the moment of invocation.{fn-org25}
```
Comment (ftomassetti) [inline bulk/repo-access-api.adoc:13]: Can the user specify that they want to use a different version of the API or is the latest one always used?

File: metametamodel/metametamodel.adoc [hunk 2]
```diff
@@ -6,7 +6,7 @@ include::../shared/issue-footnotes.adoc[]
 :sectnums:
 
 
-In this document we describe the Meta-Metamodel to be used by LionWeb.
+In this document we describe the Meta-Metamodel to be used by LionWeb *version 2024.1*.
```
Comment (ftomassetti) [inline metametamodel/metametamodel.adoc:9]: Should we then rename this file? Should we keep side by side a version of the document describing version 2023.1?

File: metametamodel/metametamodel.adoc [hunk 3]
```diff
@@ -378,13 +378,74 @@ An EnumerationLiteral is similar to Ecore’s `EEnumLiteral` and to MPS’ `Enum
...
+An EnumerationLiteral is a <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
 
 .Constraints
 Each EnumerationLiteral must have a unique `name` within the Enumeration.
 
 Each EnumerationLiteral belongs to one and only one Enumeration.
 
+[[StructuredDatatype]]
+==== StructuredDatatype
+Represents a collection of named instances of <<PrimitiveType, Primitive Types>>.
```
Comment (ftomassetti) [inline metametamodel/metametamodel.adoc:390]: I would add a comment to explain the difference from a Concept. For me the main difference is in the identify: instances of StructuredDatatype have none. They are just values, and if two StructuredDatatype have the same values for the same fields they are interchangeable, while the same is not true for nodes

File: metametamodel/metametamodel.adoc [hunk 4]
```diff
@@ -378,13 +378,74 @@ An EnumerationLiteral is similar to Ecore’s `EEnumLiteral` and to MPS’ `Enum
...
+A StructuredDatatype contains <<Field, Fields>> in its `fields` containment.
+It also represents the <<namespaces, namespace>> for the fields.
+
+A StructuredDatatype has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
+
+A StructuredDatatype is a <<DataType>> (as it can be used as <<Property.type, type>> of a <<Property>>).
+It is indirectly a <<LanguageEntity>> (as it is a top level element in a <<Language>>), an <<INamed>> (as it has a name) and a <<IKeyed>> (as it has a key).
+
+.Constraints
+M1 Instances of a StructuredDatatype SHOULD be immutable.
```
Comment (ftomassetti) [inline metametamodel/metametamodel.adoc:414]: Do we care if they are immutable or not? 
If they have no identity (i.e., we cannot refer to them) if someone modify a certain instance or create a new one , when passing them around the behavior remains the same, I think

File: metametamodel/metametamodel.adoc [hunk 5]
```diff
@@ -378,13 +378,74 @@ An EnumerationLiteral is similar to Ecore’s `EEnumLiteral` and to MPS’ `Enum
...
+.EMF & MPS equivalent
+No direct similarities.
+
+.Characteristics
+A Field has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
+
+[[Field.type, Field.type]]
+A Field refers its `type`, which is a <<DataType>>.
+
+A Field is an <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
```
Comment (ftomassetti) [inline metametamodel/metametamodel.adoc:441]: What is the multiplicity of a field?

File: roadmap/roadmap.adoc [hunk 6]
```diff
@@ -156,6 +158,13 @@ Model-to-model generators would be implemented as <<{refarch}.adoc#processor, Pr
 [[versions]]
 == Versions
 
+[[release-2024.1]]
+[[version-2024.1]]
+=== 2024.1
+* Specified <<{bulk}.adoc#bulk, Bulk Repository Access API>>
```
Comment (ftomassetti) [inline roadmap/roadmap.adoc:164]: Was that not started under 2023-1?

File: bulk/repo-access-api.adoc (hunk 1 above)
Comment (enikao) [reply bulk/repo-access-api.adoc:13]: Not decided yet. We possibly change future versions of the spec, so I want to make sure the document contains the version.

File: metametamodel/metametamodel.adoc (hunk 2 above)
Comment (enikao) [reply metametamodel/metametamodel.adoc:9]: There is a fixed version for 2023.1 in the folder with the same name. This is the _current_ version.

File: metametamodel/metametamodel.adoc (hunk 4 above)
Comment (enikao) [reply metametamodel/metametamodel.adoc:414]: As far as I remember, we've decided to follow the rationale in #265: They SHOULD be immutable, so implementations can assume that and apply optimizations. It doesn't matter for the protocol itself.

File: roadmap/roadmap.adoc (hunk 6 above)
Comment (enikao) [reply roadmap/roadmap.adoc:164]: 2023.1 was released some time ago. It will never ever change.

File: metametamodel/metametamodel.adoc (hunk 5 above)
Comment (enikao) [reply metametamodel/metametamodel.adoc:441]: Explained at the constraints of StructuredDatatype. I was unsure where to include it, as it relates to both of them.

File: metametamodel/metametamodel.adoc (hunk 2 above)
Comment (ftomassetti) [reply metametamodel/metametamodel.adoc:9]: Makes sense

File: metametamodel/metametamodel.adoc (hunk 3 above)
Comment (enikao) [reply metametamodel/metametamodel.adoc:390]: @dslmeinte to adjust

Comment (dslmeinte) [reply metametamodel/metametamodel.adoc:390]: See https://github.com/LionWeb-io/specification/pull/282 / https://github.com/LionWeb-io/specification/pull/282/commits/b527aa1158789b545abfd4069189cc67f933d4d5

**Pull request merged**